main/src/ca/uwaterloo/flix/util/TwoList.scala

```diff
@@ -35,6 +35,14 @@ case class TwoList[T](x: T, y: T, rest: List[T]) {
     */
   def forall(f: T => Boolean): Boolean = f(x) && f(y) && rest.forall(f)
 
+  /**
+   * Folds `f` over the elements of `this` list.
+   */
+  @inline
+  def foldRight[S](z: S)(f: (T, S) => S): S = {
```

Comment:
Is there a reason to foldRight instead of left? Left would be tail-recursive, I think.

Reply:
To preserve the order of constraints